Ingester/Distributor: Add support for exporting cost attribution metrics (#10269)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

# (experimental) Enables optimized marshaling of timeseries.
# CLI flag: -timeseries-unmarshal-caching-optimization-enabled
[timeseries_unmarshal_caching_optimization_enabled: <boolean> | default = true]

# (experimental) Time interval at which inactive cost attributions are evicted
# from the counter, ensuring they are not included in the cost attribution
# cardinality per user limit.
# CLI flag: -cost-attribution.eviction-interval
[cost_attribution_eviction_interval: <duration> | default = 20m]

# (experimental) Defines a custom path for the registry. When specified, Mimir

> # will expose cost attribution metrics through this custom path, if not

Reviewer: ```suggestion
# exposes cost attribution metrics through this custom path. If not
```

> # specified, cost attribution metrics won't be exposed.

Reviewer: ```suggestion
# specified, cost attribution metrics aren't exposed.
```

Reviewer: We avoid using future tense in the docs.

# CLI flag: -querier.active-series-results-max-size-bytes
[active_series_results_max_size_bytes: <int> | default = 419430400]

> # (experimental) Defines labels for cost attribution, applied to metrics like

Reviewer: ```suggestion
# (experimental) Defines labels for cost attribution. Applies to metrics like
```

> # cortex_distributor_attributed_received_samples_total. Set to an empty string
> # to disable. Example: 'team,service' will produce metrics such as

Reviewer: ```suggestion
# cortex_distributor_attributed_received_samples_total. To disable, set to an empty string.
# For example, 'team,service' produces metrics such as
```

# cortex_distributor_attributed_received_samples_total{team='frontend',
# service='api'}.
# CLI flag: -validation.cost-attribution-labels
[cost_attribution_labels: <string> | default = ""]

# (experimental) Maximum number of cost attribution labels allowed per user.
# CLI flag: -validation.max-cost-attribution-labels-per-user
[max_cost_attribution_labels_per_user: <int> | default = 2]

# (experimental) Maximum cardinality of cost attribution labels allowed per
# user.
# CLI flag: -validation.max-cost-attribution-cardinality-per-user
[max_cost_attribution_cardinality_per_user: <int> | default = 10000]

# (experimental) Cooldown period for cost attribution labels. Specifies the
# duration the cost attribution remains in overflow before attempting a reset.

> # If the cardinality remains above the limit after this period, the system will
> # stay in overflow mode and extend the cooldown. Setting this value to 0

Reviewer: ```suggestion
# If the cardinality remains above the limit after this period, the system
# stays in overflow mode and extends the cooldown. Setting this value to 0
```

Author: all doc updates addressed in eebd0288658ce4e2c9447d89f5efc8f96e0878ee

# disables the cooldown, causing the system to continuously check whether the

> # cardinality has dropped below the limit. A reset will occur once the
> # cardinality falls below the limit.

Reviewer: ```suggestion
# cardinality has dropped below the limit. A reset occurs when the
# cardinality falls below the limit.
```
